src/comgr.cpp

```diff
@@ -0,0 +1,503 @@
+/// (minimal compiler abstraction layer).
+namespace compiler {
+
+#if COMPILER_LC
+namespace lc {
+#define OCL_EARLY_INLINE 1
+
+static void AddOcl20CompilerOptions(OptionList& list)
+{
+    list.push_back("-cl-kernel-arg-info");
+    list.push_back("-D__IMAGE_SUPPORT__=1");
+    list.push_back("-D__OPENCL_VERSION__=200");
+#if OCL_EARLY_INLINE
+    list.push_back("-mllvm");
+    list.push_back("-amdgpu-early-inline-all");
+#endif
+    list.push_back("-mllvm");
+    list.push_back("-amdgpu-prelink");
+    list.push_back("-mwavefrontsize64"); // gfx1000+ WAVE32 mode: always disabled.
+    list.push_back("-mcumode");          // gfx1000+ WGP mode: always disabled.
+    list.push_back("-O3");
+
+    // It seems like these options are used only in codegen.
+    // However it seems ok to pass these to compiler.
+    if(!miopen::IsEnabled(MIOPEN_DEBUG_SRAM_EDC_DISABLED{}))
+        list.push_back("-msram-ecc");
+    else
+        list.push_back("-mno-sram-ecc");
+    list.push_back("-mllvm");
+    list.push_back("-amdgpu-internalize-symbols");
+}
+
+/// These are produced for offline compiler and not necessary at least
+/// (or even can be harmful) for building via comgr layer.
+///
+/// \todo Produce proper options in, er, proper places, and get rid of this.
+static void RemoveSuperfluousOptions(OptionList& list)
+{
+    list.erase(remove_if(list.begin(),
+                         list.end(),
+                         [&](const auto& option) { return StartsWith(option, "-mcpu="); }),
+               list.end());
+}
+
+/// \todo Get list of supported isa names from comgr and select.
+static std::string GetIsaName(const std::string& device)
+{
+    const char* const ecc_suffix = (!miopen::IsEnabled(MIOPEN_DEBUG_SRAM_EDC_DISABLED{}) &&
+                                    (device == "gfx906" || device == "gfx908"))
+                                       ? "+sram-ecc"
+                                       : "";
+    return {"amdgcn-amd-amdhsa--" + device + ecc_suffix};
+}
+
+/// \todo Handle "-cl-fp32-correctly-rounded-divide-sqrt".
+
+} // namespace lc
+#undef OCL_EARLY_INLINE
+#endif // COMPILER_LC
+
+} // namespace compiler
+
+struct EcInfoNone
+{
+};
+static const EcInfoNone NoInfo;
+static inline std::string to_string(const EcInfoNone&) { return {}; }
+static inline std::string to_string(const std::string& v) { return {v}; }
+static inline std::string to_string(const bool& v) { return v ? "true" : "false"; }
+static inline auto to_string(const std::size_t& v) { return std::to_string(v); }
+
+#define CASE_TO_STRING(id) \
+    case id: return #id
+
+/// \todo Request comgr to expose this stuff via API.
+static std::string to_string(const amd_comgr_language_t val)
+{
+    switch(val)
+    {
+        CASE_TO_STRING(AMD_COMGR_LANGUAGE_NONE);
+        CASE_TO_STRING(AMD_COMGR_LANGUAGE_OPENCL_1_2);
+        CASE_TO_STRING(AMD_COMGR_LANGUAGE_OPENCL_2_0);
+        CASE_TO_STRING(AMD_COMGR_LANGUAGE_HC);
+        CASE_TO_STRING(AMD_COMGR_LANGUAGE_HIP);
```

Comment:
Use `MIOPEN_LOG_ENUM` for this instead of creating new macros.

Reply:
Done.